test/functional/terminal/ex_terminal_spec.lua

```diff
@@ -48,6 +49,31 @@ describe(':terminal', function()
     eq(2, eval("line('.')"))  -- cursor stays where we put it
   end)
 
+  it('does not crash when callback fires after terminal closes', function()
+    screen:detach()
+    screen = thelpers.screen_setup(0, '["'..helpers.nvim_prog..'", "-u", "NONE", "-i", "NONE", "-N"]')
```

Comment:
Right, I meant `'-n'`, since I didn't want to type out `'--cmd', 'set noswapfile'`.